### ISSUE TITLE
Serialiser_Engine: Add BSON deep clone method

### ISSUE THREAD

**Issue by IsakNaslundBh** [BHoM/BHoM_Engine#768]
Could we add a simple BSON Deep clone method to the serialisation_Engine?

something in the lines of:

```

namespace BH.Engine.Serialiser
{
    public static partial class Modify
    {
        /*******************************************/
        /**** Public Methods                    ****/
        /*******************************************/

        public static object BsonDeepClone(this object obj)
        {
            return Convert.FromBson(obj.ToBson());
        }

        /*******************************************/
    }
}

```

to simply get out a full clone of an object.

**Comment by IsakNaslundBh** [comment]
Yeah, agreed this is not the most efficient way to do it. Only reason for the suggestion is that it is one line of code to implement to unlock deep cloning, but if we have a better way in the works I can wait for that.

Was the idea to simply use the force-net Deep cloner you linked to, or write something up ourselfs?

**Comment by adecler** [comment]
The license of DeepCloner is MIT so I don't see any problem using it directly.

**Comment by IsakNaslundBh** [comment]
Thanks @adecler 

Closing this as better solved by #175
